sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -234,6 +234,21 @@ object DottyPlugin extends AutoPlugin {
       // bootclasspath, and instead have scala-library and dotty-library on the
       // compiler classpath. This means that user code could shadow symbols
       // from these jars but we can live with that for now.
+
+      // sbt crazy scoping rules mean that when we override `classpathOptions`
+      // below we also override `classpathOptions in console` which is normally
+      // set in https://github.com/sbt/sbt/blob/b6f02b9b8cd0abb15e3d8856fd76b570deb1bd61/main/src/main/scala/sbt/Defaults.scala#L503,
+      // this breaks `sbt console` in Scala 2 projects.
+      // There seems to be no way to avoid stomping over task-scoped settings,
+      // so we need to manually set `classpathOptions in console` to something sensible,
+      // ideally this would be "whatever would be set if this plugin was not enabled",
+      // but I can't find a way to do this, so we default to whatever is set in ThisBuild.
+      classpathOptions in console := {
+        if (isDotty.value)
+          classpathOptions.value // The Dotty REPL doesn't require anything special on its classpath
+        else
+          (classpathOptions in console in ThisBuild).value
```

Comment:
You can preserve existing definitions of `classpath in console` by using the `.?` combinator:

```suggestion
          (classpathOptions in console).?.value.getOrElse((classpathOptions in console in ThisBuild).value)
```

Those would be project-level settings that come before `DottyPlugin`, which might be niche/inexistent.

Btw the underlying sbt issue here is https://github.com/sbt/sbt/issues/2899, which I've been investigating.

Reply:
Wow, I had never heard of `?` before! However it doesn't seem to be enough, `(classpathOptions in console).?.value is never `None` when I run it, instead it seems to always be equal to `Some((classpathOptions in console).value)` which ends up using the `classpathOptions` definition below.

Reply:
Ah, yeah.  It's not undefined as it just falls back to the problematically-redefined project-level `classpathOptions` setting.

Worth a try.

Reply:
What's wrong with `(classpathOptions in console).value`? That will pick up project-level if it's there, and it will delegate to `ThisBuild` when it isn't automatically.

Reply:
The problem is that it picks up the `classpathOptions` set just below.